src/main/java/fr/openent/competences/service/impl/DefaultElementProgramme.java

```diff
@@ -70,6 +73,26 @@ public void getElementProgramme(Long idPeriode, String idMatiere, String idClass
         sql.prepared(query.toString(), values, SqlResult.validUniqueResultHandler(handler));
     }
 
+    public Future<JsonObject> getElementProgramme(Long idPeriode, String idMatiere, String idClasse){
+        Promise<JsonObject> promiseElementProgramme = Promise.promise();
+        StringBuilder query = new StringBuilder();
+        JsonArray values = new fr.wseduc.webutils.collections.JsonArray();
+
+        query.append("SELECT * ")
+                .append("FROM "+ Competences.COMPETENCES_SCHEMA +".element_programme ")
+                .append("WHERE "+ Competences.COMPETENCES_SCHEMA +".element_programme.id_classe = ? ")
+                .append("AND "+ Competences.COMPETENCES_SCHEMA +".element_programme.id_periode = ? ")
+                .append("AND "+ Competences.COMPETENCES_SCHEMA +".element_programme.id_matiere = ? ");
+
+        values.add(idClasse);
+        values.add(idPeriode);
+        values.add(idMatiere);
+        sql.prepared(query.toString(), values,
+                SqlResult.validUniqueResultHandler(FutureHelper.handlerJsonObject(promiseElementProgramme, "[DefaultElementProgramme] : getElementProgramme :")));
+
+        return promiseElementProgramme.future();
+    }
```

Comment:
Même principe par rapport à mon commentaire au-dessus : utilise la méthode déjà existante pour polymorpher ta méthode `getElementProgramme` que tu veux faire retourner une future

Exemple usage :

```suggestion
     public Future<JsonObject> getElementProgramme(Long idPeriode, String idMatiere, String idClasse){
        Promise<JsonObject> promiseElementProgramme = Promise.promise();
        getElementProgramme(idPeriode, idMatiere, idClasse, event -> {
            if (event.isLeft()) {
                // instancier une classe logger dans cette classe pour ajouter log.error ?
                // private final Logger log = LoggerFactory.getLogger(DefaultElementProgramme.class);
                log.error(String.format("[Competences@%s::getElementProgramme] Error during sql request: %s.",
                        this.getClass().getSimpleName(), event.left().getValue()));
                promiseElementProgramme.fail(event.left().getValue());
            } else {
                promiseElementProgramme.complete(event.right().getValue());
            }
        });

        return promiseElementProgramme.future();
    }
```